### PR TITLE
MUZIMA-440, MUZIMA-393

### PR DESCRIPTION
**Pull request by mssavai**

### CODE REVIEW

File: src/main/java/com/muzima/service/MuzimaSyncService.java [hunk 1]
```diff
@@ -712,4 +832,44 @@ private void checkChangeInPatientId(Patient localPatient, Patient patientFromSer
...
         return result;
     }
+
+    public int[] downloadSetupConfigurations(){
+        int[] result = new int[2];
+        try {
+            List<SetupConfiguration> setupConfigurations = setupConfigurationController.downloadAllSetupConfigurations();
+            result[0] = SyncStatusConstants.SUCCESS;
+            result[1] = setupConfigurations.size();
+            System.out.println("Setup Configs downloaded: "+setupConfigurations.size());
```
Comment (sthaiya) [inline src/main/java/com/muzima/service/MuzimaSyncService.java]: Replace with logger

File: src/main/java/com/muzima/service/MuzimaSyncService.java [hunk 2]
```diff
@@ -712,4 +832,44 @@ private void checkChangeInPatientId(Patient localPatient, Patient patientFromSer
...
+    public int[] downloadSetupConfigurationTemplate(String uuid){
+        int[] result = new int[2];
+        try {
+            SetupConfigurationTemplate setupConfigurationTemplate =
+                    setupConfigurationController.downloadSetupConfigurationTemplate(uuid);
+            result[0] = SyncStatusConstants.SUCCESS;
+            if(setupConfigurationTemplate != null) {
+                result[1] = 1;
+            }
+            System.out.println("MuzimaSyncService: "+setupConfigurationTemplate.getConfigJson());
```
Comment (sthaiya) [inline src/main/java/com/muzima/service/MuzimaSyncService.java]: Replace with logger

File: src/main/java/com/muzima/view/patients/PatientsListActivity.java [hunk 3]
```diff
@@ -281,6 +307,36 @@ protected void onActivityResult(int requestCode, int resultCode, Intent dataInte
     @Override
     public void onBackPressed(){
         patientAdapter.cancelBackgroundTask();
+        listView.clearFocus();
         super.onBackPressed();
+        if(getCurrentFocus() != null) {
+            System.out.println("Current focusssssssss: " + getCurrentFocus().toString());
```
Comment (sthaiya) [inline src/main/java/com/muzima/view/patients/PatientsListActivity.java]: Do we need to remove this?

File: src/main/java/com/muzima/view/patients/PatientsListActivity.java [hunk 4]
```diff
@@ -281,6 +307,36 @@ protected void onActivityResult(int requestCode, int resultCode, Intent dataInte
...
+    @Override
+    public void onKeyboardClosed() {
+        fabSearchButton.setVisibility(View.VISIBLE);
+       // fabSearchButton.setFocusable(true);
+       // boolean f=fabSearchButton.requestFocus();
+        listView.clearChoices();
+        if(getCurrentFocus() != null) {
+            System.out.println("Current focus: " + getCurrentFocus().toString());
+        } else {
+            System.out.println("Current focus: null");
```
Comment (sthaiya) [inline src/main/java/com/muzima/view/patients/PatientsListActivity.java]: Do we need to remove this?

File: src/main/java/com/muzima/view/setupconfiguration/GuidedConfigurationWizardActivity.java [hunk 5]
```diff
@@ -0,0 +1,620 @@
...
+            @Override
+            protected void onPreExecute() {
+                downloadLocationsLog.setSetupAction(getString(R.string.info_location_download));
+                onQueryTaskStarted();
+            }
+            @Override
+            protected int[] doInBackground(Void... voids) {
+                List<String> uuids = extractLocationsUuids();
+                if (!uuids.isEmpty())
+
```
Comment (sthaiya) [inline src/main/java/com/muzima/view/setupconfiguration/GuidedConfigurationWizardActivity.java]: Add TODO: or inverse the if to remove the blank then-clause

File: src/main/java/com/muzima/view/setupconfiguration/GuidedConfigurationWizardActivity.java [hunk 6]
```diff
@@ -0,0 +1,620 @@
...
+            @Override
+            protected void onPreExecute() {
+                downloadProvidersLog.setSetupAction(getString(R.string.info_provider_download));
+                onQueryTaskStarted();
+            }
+            @Override
+            protected int[] doInBackground(Void... voids) {
+                List<String> uuids = extractProvidersUuids();
+                if (!uuids.isEmpty())
+
```
Comment (sthaiya) [inline src/main/java/com/muzima/view/setupconfiguration/GuidedConfigurationWizardActivity.java]: same as above

File: src/main/java/com/muzima/view/setupconfiguration/GuidedConfigurationWizardActivity.java [hunk 7]
```diff
@@ -0,0 +1,620 @@
...
+            @Override
+            protected void onPreExecute() {
+                downloadConceptsLog.setSetupAction(getString(R.string.info_concept_download));
+                onQueryTaskStarted();
+            }
+            @Override
+            protected int[] doInBackground(Void... voids) {
+                List<String> uuids = extractConceptsUuids();
+                if (!uuids.isEmpty())
+
```
Comment (sthaiya) [inline src/main/java/com/muzima/view/setupconfiguration/GuidedConfigurationWizardActivity.java]: This too

File: src/main/res/layout/activity_login.xml [hunk 8]
```diff
@@ -23,7 +23,7 @@
                     android:id="@+id/serverUrl"
                     style="@style/LoginInfoEditText"
                     android:inputType="textUri"
-                    android:text="http://demo1.muzima.org"
+                    android:text="http://demo2.muzima.org"
```
Comment (sthaiya) [inline src/main/res/layout/activity_login.xml]: This needs to be a string resource "default_server"

File: src/main/res/xml/preference.xml [hunk 9]
```diff
@@ -3,7 +3,7 @@
 
     <PreferenceCategory android:title="@string/header_openmrs_category">
         <EditTextPreference
-                android:defaultValue="http://demo1.muzima.org"
+                android:defaultValue="http://demo2.muzima.org"
```
Comment (sthaiya) [inline src/main/res/xml/preference.xml]: We need to make this a string resource "default_server"

**Comment by mssavai** [comment]
@sthaiya  The changes are now effected. Thanks.

**Pull request merged**